### PR TITLE
microsoft-edge: Mirror google-chrome derivation

### PR DESCRIPTION
**Pull request by rhysmdnz**
Changed the edge derivation to be closer to a copy of the google chrome derivation. Should make it easier to follow what is going on in that. Most of the changes that need to be made to edge tend to get done in chrome first.

Sourced from: https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/go/google-chrome/package.nix

Have also removed beta and dev for the same reasons they were dropped in chrome. no one's testing them.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by rhysmdnz** [comment]
This fixes a crash in recent versions of edge upon opening the dev tools

File: pkgs/by-name/mi/microsoft-edge/package.nix [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    changelog = "https://learn.microsoft.com/en-us/deployedge/microsoft-edge-relnote-stable-channel";
+    license = licenses.unfree;
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+    platforms = [ "x86_64-linux" ];
+    maintainers = with maintainers; [
+      zanculmarktum
+      kuwii
+      rhysmdnz
+    ];
+  };
```
Comment (gepbird) [inline pkgs/by-name/mi/microsoft-edge/package.nix:288]: Please set `meta.mainProgram`, order `meta` attrs alphabetically and remove `with lib;`

File: pkgs/by-name/mi/microsoft-edge/package.nix [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    ++ lib.optional pulseSupport libpulseaudio
+    ++ lib.optional libvaSupport libva
+    ++ [
+      gtk3
+      gtk4
+    ];
+in
+
+stdenv.mkDerivation rec {
+  inherit pname;
```
Comment (gepbird) [inline pkgs/by-name/mi/microsoft-edge/package.nix:174]: This is the only place `pname` is used
```suggestion
  pname = "microsoft-edge";
```

File: pkgs/by-name/mi/microsoft-edge/package.nix [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+  addDriverRunpath,
+  undmg,
+
+  # Edge Specific
+  libuuid,
+}:
+
+let
+
+  pname = "microsoft-edge";
```
Comment (gepbird) [inline pkgs/by-name/mi/microsoft-edge/package.nix:103]:
```suggestion
```

File: pkgs/by-name/mi/microsoft-edge/package.nix [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+  '';
+
+  rpath = lib.makeLibraryPath deps + ":" + lib.makeSearchPathOutput "lib" "lib64" deps;
+  binpath = lib.makeBinPath deps;
+
+  installPhase = ''
+    runHook preInstall
+
+    appname=msedge
+    dist=stable
```
Comment (gepbird) [inline pkgs/by-name/mi/microsoft-edge/package.nix:215]: Maybe the `dist` should be hard coded?
```suggestion
```

**Review by gepbird — CHANGES_REQUESTED** [review]
Thanks for your PR! I agree with the removal of the non-stable channels. Many of the refactors look cleaner, but I'm not experienced with this derivation to judge all those dependency and build script changes. Review from another maintainer would be very welcome. I'll only leave some nits.

Next time when you make big refactors, please do it in multiple commits, For example a by-name migration and a many lines of formatting changes should be in separate commits. It's harder to review when the diff only says removed every line of microsoft-edge/browser.nix, added every line of microsoft-edge/package.nix.

**Comment by gepbird** [comment]
cc @zanculmarktum @kuwii

**Comment by Daholli** [comment]
can we also move up to 130.0.2849.1 since that seems to be the new stable now

File: pkgs/by-name/mi/microsoft-edge/package.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/by-name/mi/microsoft-edge/package.nix:215]: that would simplify the code

File: pkgs/by-name/mi/microsoft-edge/package.nix [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+  flac,
+  harfbuzz,
+  icu,
+  libopus,
+  libpng,
+  snappy,
+  speechd-minimal,
+  ## Gentoo
+  bzip2,
+  libcap,
```
Comment (SuperSandro2000) [inline pkgs/by-name/mi/microsoft-edge/package.nix:80]: We should add more useful comments than just copying other distros

File: pkgs/by-name/mi/microsoft-edge/package.nix [hunk 6]
```diff
@@ -0,0 +1,289 @@
+{
+  fetchurl,
+  lib,
+  makeWrapper,
+  patchelf,
+  stdenv,
+  stdenvNoCC,
```
[the remark this answers is not in the document]
Comment (rhysmdnz) [reply pkgs/by-name/mi/microsoft-edge/package.nix]: Removed the unused deps from chromes darwin build

**Comment by rhysmdnz** [comment]
> Thanks for your PR! I agree with the removal of the non-stable channels. Many of the refactors look cleaner, but I'm not experienced with this derivation to judge all those dependency and build script changes. Review from another maintainer would be very welcome. I'll only leave some nits.
> 
> Next time when you make big refactors, please do it in multiple commits, For example a by-name migration and a many lines of formatting changes should be in separate commits. It's harder to review when the diff only says removed every line of microsoft-edge/browser.nix, added every line of microsoft-edge/package.nix.

Agreed, but there's not really a smaller cleaner diff. This was more a copy over https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/go/google-chrome/package.nix and make the relevant changes to get it back to edge. than it was change the edge derivation.

I've split it up a bit more now.

**Comment by Daholli** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 347836`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>microsoft-edge</li>
  </ul>
</details>

**Review by gepbird — APPROVED** [review]

**Pull request merged**